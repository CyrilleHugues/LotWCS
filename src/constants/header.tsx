export const HEADER_SET_NAME = 'HEADER_SET_NAME';
export type HEADER_SET_NAME = typeof HEADER_SET_NAME;

export const HEADER_SET_CONCEPT = 'HEADER_SET_CONCEPT';
export type HEADER_SET_CONCEPT = typeof HEADER_SET_CONCEPT;
