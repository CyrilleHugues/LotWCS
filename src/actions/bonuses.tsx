import * as constants from '../constants/bonuses';
import { IBonus,
  IBonusDestiny, IBonusEntanglement, IBonusOneAmongN,
  IBonusSkillRank, IBonusSpeciality, IBonusStartingChi,
} from '../types/bonuses';

export function destiny(value: number): IBonusDestiny {
    return { type: constants.BONUS_DESTINY, value };
}

export function entanglement(value: number): IBonusEntanglement {
  return { type: constants.BONUS_ENTANGLEMENT, value };
}

export function startingChi(value: number): IBonusStartingChi {
  return { type: constants.BONUS_STARTING_CHI, value };
}

export function oneAmongN(bonuses: IBonus[]): IBonusOneAmongN {
  return { type: constants.BONUS_ONE_AMONG_N, bonuses };
}

export function skillRank(skill: string): IBonusSkillRank {
  return { type: constants.BONUS_SKILL_RANK, skill };
}

export function speciality(skill: string, specialityName: string): IBonusSpeciality {
  return { type: constants.BONUS_SPECIALITY, skill, speciality: specialityName };
}
