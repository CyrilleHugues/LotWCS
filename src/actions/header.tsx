import * as constants from '../constants/header';

export interface IHeaderSetName {
  type: constants.HEADER_SET_NAME;
  name: string;
}

export interface IHeaderSetConcept {
  type: constants.HEADER_SET_CONCEPT;
  concept: string;
}

export interface IHeaderSetArchetype {
  type: constants.HEADER_SET_ARCHETYPE;
  archetype: string;
}

export interface IHeaderSetRank {
  type: constants.HEADER_SET_RANK;
  rank: string;
}

export type IHeaderAction = IHeaderSetName | IHeaderSetConcept | IHeaderSetArchetype | IHeaderSetRank;

export function headerSetName(s: string): IHeaderSetName {
  return {
    type: constants.HEADER_SET_NAME,
    name: s
  };
}

export function headerSetConcept(s: string): IHeaderSetConcept {
  return {
    type: constants.HEADER_SET_CONCEPT,
    concept: s
  };
}

export function headerSetArchetype(s: string): IHeaderSetArchetype {
  return {
    type: constants.HEADER_SET_ARCHETYPE,
    archetype: s
  };
}

export function headerSetRank(s: string): IHeaderSetRank {
  return {
    type: constants.HEADER_SET_RANK,
    rank: s
  };
}
