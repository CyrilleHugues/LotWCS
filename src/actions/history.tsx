import * as constants from '../constants/history';

export interface IHistoryDelete {
  type: constants.HISTORY_DELETE;
  id: number;
}

export type IHistoryAction = IHistoryDelete;

export function historyDelete(id: number): IHistoryDelete {
  return {
    type: constants.HISTORY_DELETE,
    id: id,
  };
}
