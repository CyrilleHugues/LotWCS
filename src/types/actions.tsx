import { IInitialStateAction } from '../actions/initial';
import { IHeaderAction } from '../actions/header';
import { ISkillAction } from '../actions/skills';

export type IAction = IInitialStateAction | IHeaderAction | ISkillAction;
