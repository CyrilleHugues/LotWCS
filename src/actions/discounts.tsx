import * as constants from '../constants/discounts';
import { skills } from '../data/skills';
import { IDiscountSkill } from '../types/discounts';
import { IDataSkill } from '../types/skills';

export function discountSkillFactory(value: number, subset?: string[]): IDiscountSkill {
  if (subset) {
    return { skills: subset, type: constants.DISCOUNT_SKILL, value, };
  } else {
    return {
      skills: skills.map((s: IDataSkill) => s.name),
      type: constants.DISCOUNT_SKILL,
      value,
    };
  }
}
