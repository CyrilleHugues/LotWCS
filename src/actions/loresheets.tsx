import * as constants from '../constants/loresheets';
import { ICost } from '../types/costs';

export interface ILoresheetOpen {
  type: constants.LORESHEET_OPEN;
  uid: string;
  cost: ICost;
}

export interface ILoresheetBuyOption {
  type: constants.LORESHEET_BUY_OPTION;
  lsUid: string;
  uid: string;
  cost: ICost;
}

export type ILoresheetAction = ILoresheetOpen | ILoresheetBuyOption;

export function open(uid: string, cost: ICost): ILoresheetOpen {
  return { cost, type: constants.LORESHEET_OPEN, uid };
}

export function buyOption(lsUid: string, uid: string, cost: ICost): ILoresheetBuyOption {
  return { cost, lsUid, type: constants.LORESHEET_BUY_OPTION, uid };
}
