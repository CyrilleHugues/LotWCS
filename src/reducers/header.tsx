import { IHeaderAction } from '../actions/header';
import { IStoreState } from '../types/state';
import * as constants from '../constants/header';
import { pushToHistory } from './history';

export function headerReducer(oldState: IStoreState, action: IHeaderAction): IStoreState {
  switch (action.type) {
    case constants.HEADER_SET_NAME:
      return oldState.withMutations(state => {
        state.set('name', action.name);
        pushToHistory(state, action);
      });
    case constants.HEADER_SET_CONCEPT:
      return oldState.withMutations(state => {
        state.set('concept', action.concept);
        pushToHistory(state, action);
      });
    case constants.HEADER_SET_RANK:
      return oldState.withMutations(state => {
        state.set('rank', action.rank );
        state.set('rankModified', true);
        pushToHistory(state, action);
      });
    case constants.HEADER_SET_ARCHETYPE:
      return oldState.withMutations(state => {
        state.set('archetype', action.archetype );
        state.set('archetypeModified', true);
        pushToHistory(state, action);
      });
    default:
  }
  return oldState;
}
