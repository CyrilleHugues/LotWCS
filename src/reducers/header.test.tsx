import * as Immutable from 'immutable';
import * as actions from '../actions/header';
import { headerReducer } from './header';

const initialState = Immutable.fromJS({
  name: 'Bob',
  concept: 'The Bobbest',
  stuff: 'Data',
});

it('HEADER_SET_NAME', () => {
  expect( headerReducer(initialState, actions.headerSetName('Robert') ) ).toMatchSnapshot();
});

it('HEADER_SET_CONCEPT', () => {
  expect( headerReducer(initialState, actions.headerSetConcept('Sir Robert') ) ).toMatchSnapshot();
});
