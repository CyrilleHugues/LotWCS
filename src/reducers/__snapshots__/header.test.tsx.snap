// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADER_SET_ARCHETYPE 1`] = `
Object {
  "archetype": "warrior",
  "archetypeModified": true,
  "concept": "No Concept",
  "history": Array [
    Object {
      "type": "INITIAL_STATE",
    },
    Object {
      "archetype": "warrior",
      "type": "HEADER_SET_ARCHETYPE",
    },
  ],
  "name": "No Name",
  "rank": "",
  "rankModified": false,
  "skills": Array [
    Object {
      "name": "Awareness",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Confidence",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Crafting",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Finesse",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Hardiness",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Inspire",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Learning",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Medicine",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Might",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Perform",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Politics",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Ride",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Stealth",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Survival",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Tactics",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Wu Wei",
      "specialities": Immutable.List [
],
      "value": 0,
    },
  ],
}
`;

exports[`HEADER_SET_CONCEPT 1`] = `
Object {
  "archetype": "",
  "archetypeModified": false,
  "concept": "Sir Robert",
  "history": Array [
    Object {
      "type": "INITIAL_STATE",
    },
    Object {
      "concept": "Sir Robert",
      "type": "HEADER_SET_CONCEPT",
    },
  ],
  "name": "No Name",
  "rank": "",
  "rankModified": false,
  "skills": Array [
    Object {
      "name": "Awareness",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Confidence",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Crafting",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Finesse",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Hardiness",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Inspire",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Learning",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Medicine",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Might",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Perform",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Politics",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Ride",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Stealth",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Survival",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Tactics",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Wu Wei",
      "specialities": Immutable.List [
],
      "value": 0,
    },
  ],
}
`;

exports[`HEADER_SET_NAME 1`] = `
Object {
  "archetype": "",
  "archetypeModified": false,
  "concept": "No Concept",
  "history": Array [
    Object {
      "type": "INITIAL_STATE",
    },
    Object {
      "name": "Robert",
      "type": "HEADER_SET_NAME",
    },
  ],
  "name": "Robert",
  "rank": "",
  "rankModified": false,
  "skills": Array [
    Object {
      "name": "Awareness",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Confidence",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Crafting",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Finesse",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Hardiness",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Inspire",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Learning",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Medicine",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Might",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Perform",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Politics",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Ride",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Stealth",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Survival",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Tactics",
      "specialities": Immutable.List [
],
      "value": 0,
    },
    Object {
      "name": "Wu Wei",
      "specialities": Immutable.List [
],
      "value": 0,
    },
  ],
}
`;
